hot-reload.md

```diff
@@ -281,3 +284,11 @@ to:
 
 In these situations, hot reload generates a diagnostic message and fails without
 committing any changes.
+
+## How it works
+
+When hot reload is invoked, the host machine looks at the edited code since the last compilation. Each library with changed code, the application's main library, and the libraries from the main library leading to affected libraries are recompiled. In Dart 2, those libraries' Dart source code are turned into [kernel files](https://github.com/dart-lang/sdk/tree/master/pkg/kernel) and sent to the mobile device's Dart VM.
+
+The Dart VM re-loads all libraries from the new kernel file. So far no code is re-executed.
+
+The hot reload mechanism then causes the Flutter framework to trigger a rebuild/re-layout/repaint of all widgets and render objects.
```

Comment:
"of all widgets and.." => "of all affected widgets and..."  ?

Reply:
I don't want to give the impression that only some of the widgets are rebuilt since all widgets are rebuilt. Changed to 'all existing widgets'.